src/coreclr/gc/gc.cpp

```diff
@@ -10546,19 +10547,52 @@ int gc_heap::object_gennum_plan (uint8_t* o)
 #endif //_MSC_VER && TARGET_X86
 
 #ifdef USE_REGIONS
+void get_initial_region(int nhp, int gen, int hn, uint8_t** region_start, uint8_t** region_end)
```

Comment:
is there any particular reason why we need to calculate the location of each region instead of just storing the start/end of each region per generation per heap when we allocate them `allocate_initial_regions`? something like this (pseudo code) -

```
initial_region_info[total_geneneration_count][nhp][2];

for (int i = 0; i < nhp; i++) 
    allocate_large (&(initial_region_info[poh_generation][i][0]), 
                    &(initial_region_info[poh_generation][i][1]));

for (int i = 0; i < nhp; i++) 
    for (int gen = max_generation; gen >= 0; gen--)
        allocate_large (&(initial_region_info[gen][i][0]), 
                        &(initial_region_info[gen][i][1]));

for (int i = 0; i < nhp; i++) 
    allocate_large (&(initial_region_info[loh_generation][i][0]), 
                    &(initial_region_info[loh_generation][i][1]));
```

then `get_initial_region` can just take the gen number and heap number and return initial_region_info[gen_number][heap_number]